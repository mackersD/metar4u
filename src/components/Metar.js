import React from 'react'
import Vector from './Vector'
import MetarText from './MetarText'
import Spinner from './Spinner'

const Metar = (props) => {
  return (
    <div className="metar">
      <Spinner
        isFetching={props.isFetching}
        isFailed={props.isFailed}
      >
        <Vector
          distance={props.distance}
          direction={props.direction}
        />
        <MetarText
          text={props.rawReport}
        />
      </Spinner>
    </div>
  )
}

export default Metar
