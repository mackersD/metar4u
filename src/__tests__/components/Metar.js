import React from 'react'
import renderer from 'react-test-renderer'
import Metar from '../../components/Metar'

it('Metar - renders', () => {
  const tree = renderer.create(
    <Metar />
  )
  expect(tree).toMatchSnapshot()
})

it('Metar - fetching', () => {
  var tree = renderer.create(
    <Metar
      isFetching={true}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Metar
      isFetching={true}
      distance={20}
      direction={135}
    />
  )
  expect(tree).toMatchSnapshot()

  var text = "KAUS 091853Z 16008KT 10SM FEW065 FEW250 36/20 A3004 RMK AO2 SLP158 T03560200"
  tree = renderer.create(
    <Metar
      isFetching={true}
      rawReport={text}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Metar
      isFetching={true}
      distance={20}
      direction={135}
      rawReport={text}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('Metar - failed', () => {
  var tree = renderer.create(
    <Metar
      isFailed={true}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Metar
      isFailed={true}
      distance={20}
      direction={135}
    />
  )
  expect(tree).toMatchSnapshot()

  var text = "KAUS 091853Z 16008KT 10SM FEW065 FEW250 36/20 A3004 RMK AO2 SLP158 T03560200"
  tree = renderer.create(
    <Metar
      isFailed={true}
      rawReport={text}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Metar
      isFailed={true}
      distance={20}
      direction={135}
      rawReport={text}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('Metar - vector + report', () => {
  var text = "KAUS 091853Z 16008KT 10SM FEW065 FEW250 36/20 A3004 RMK AO2 SLP158 T03560200"
  var tree = renderer.create(
    <Metar
      distance={20}
      direction={135}
      rawReport={text}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Metar
      distance={20}
      direction={135}
    />
  )
  expect(tree).toMatchSnapshot()

  tree = renderer.create(
    <Metar
      rawReport={text}
    />
  )
  expect(tree).toMatchSnapshot()
})
