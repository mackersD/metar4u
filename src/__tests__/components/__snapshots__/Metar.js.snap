// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metar - failed 1`] = `
<div
  className="metar"
/>
`;

exports[`Metar - failed 2`] = `
<div
  className="metar"
/>
`;

exports[`Metar - failed 3`] = `
<div
  className="metar"
/>
`;

exports[`Metar - failed 4`] = `
<div
  className="metar"
/>
`;

exports[`Metar - fetching 1`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    Fetching
  </div>
</div>
`;

exports[`Metar - fetching 2`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    Fetching
  </div>
</div>
`;

exports[`Metar - fetching 3`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    Fetching
  </div>
</div>
`;

exports[`Metar - fetching 4`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    Fetching
  </div>
</div>
`;

exports[`Metar - renders 1`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    <div
      className="vector"
    >
      <div
        className="vectorArrow"
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
            "display": "inline-block",
            "msTransform": "rotate(0deg)",
            "transform": "rotate(0deg)",
          }
        }
      >
        ^
      </div>
      <div
        className="vectorDistance"
      >
        0
      </div>
    </div>
    <div
      className="metarText"
    />
  </div>
</div>
`;

exports[`Metar - vector + report 1`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    <div
      className="vector"
    >
      <div
        className="vectorArrow"
        style={
          Object {
            "WebkitTransform": "rotate(135deg)",
            "display": "inline-block",
            "msTransform": "rotate(135deg)",
            "transform": "rotate(135deg)",
          }
        }
      >
        ^
      </div>
      <div
        className="vectorDistance"
      >
        20
      </div>
    </div>
    <div
      className="metarText"
    >
      KAUS 091853Z 16008KT 10SM FEW065 FEW250 36/20 A3004 RMK AO2 SLP158 T03560200
    </div>
  </div>
</div>
`;

exports[`Metar - vector + report 2`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    <div
      className="vector"
    >
      <div
        className="vectorArrow"
        style={
          Object {
            "WebkitTransform": "rotate(135deg)",
            "display": "inline-block",
            "msTransform": "rotate(135deg)",
            "transform": "rotate(135deg)",
          }
        }
      >
        ^
      </div>
      <div
        className="vectorDistance"
      >
        20
      </div>
    </div>
    <div
      className="metarText"
    />
  </div>
</div>
`;

exports[`Metar - vector + report 3`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    <div
      className="vector"
    >
      <div
        className="vectorArrow"
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
            "display": "inline-block",
            "msTransform": "rotate(0deg)",
            "transform": "rotate(0deg)",
          }
        }
      >
        ^
      </div>
      <div
        className="vectorDistance"
      >
        0
      </div>
    </div>
    <div
      className="metarText"
    >
      KAUS 091853Z 16008KT 10SM FEW065 FEW250 36/20 A3004 RMK AO2 SLP158 T03560200
    </div>
  </div>
</div>
`;
